ruby-challenges.rb

```diff
@@ -1,32 +1,178 @@
+#--------------------||✅ Final Test Results ✅||------------------------------
+# ruby-challenges.rb
+# ✅ TEST PASSED ==> 🧪gets words that contain character
+# ✅ TEST PASSED ==> 🧪gets words that contain character
+#-------------------------------------------------------------------------------
+#*******************************************************************************
+# -------------------2) Create a method that takes in a hash and returns one 
+# array with all the hash values at their own index and in alphabetical order. 
+# No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
+#*******************************************************************************
+
+#************************** Explanation ****************************************
+# I accessed the values of the hash with the values method, then chained that 
+# with the flatten method to assign those values inside a "flat" array, then I 
+# chained that with the sort method to alphabetize the values.
+#*******************************************************************************
+def flatten_hash(hash); hash.values.flatten.sort end
+
+#---------------------------||🧪 Test Cases 🧪||--------------------------------
+# Challenge 2, Test 1    
+
+us_states = { 
+  northwest: ['Washington', 'Oregon', 'Idaho'], 
+  southwest: ['California', 'Arizona', 'Nevada'], 
+  notheast: ['Maine', 'New Hampshire', 'Vermont'] 
+}     
+
+assertEqual(flatten_hash(us_states), 
+[
+  'Arizona', 
+  'California', 
+  'Idaho', 
+  'Maine', 
+  'Nevada',
+  'New Hampshire', 
+  'Oregon', 
+  'Vermont', 
+  'Washington'
+],                            
+'flattens and organizes hash values alphabetically')      
+
+#--------------------||✅ Final Test Results ✅||------------------------------
+# ruby-challenges.rb
+# ✅ TEST PASSED ==> 🧪gets words that contain character
+# ✅ TEST PASSED ==> 🧪gets words that contain character
+# ✅ TEST PASSED ==> 🧪flattens and organizes hash values alphabetically
+#-------------------------------------------------------------------------------
+
+#*******************************************************************************
+# --------------------3a) Create a class called Bike that is initialized with a 
+# model, wheels, and current_speed. The default number of wheels is 2. The 
+# current_speed should start at 0. Create a bike_info method that returns a 
+# sentence with all the data from the bike object.
+#-------------------------------------------------------------------------------
+# -------------------3b) Add the ability to pedal faster and brake. The 
+# pedal_faster method should increase the speed by a given amount. The brake 
+# method should decrease the speed by a given amount. The bike cannot go 
+# negative speeds.
+#*******************************************************************************
+class Bike 
```

Comment:
Nice work!  Once again, really like that you are building your own test method and testing your work 👏